src/csharp/BUILD-INTEGRATION.md

```diff
@@ -440,3 +182,256 @@ protoc --plugin=protoc-gen-grpc=grpc_csharp_plugin --csharp_out=OUT_DIR \
+```
+
+## MSBuild Properties
+
+You can set some Properties in your project file or on the MSBuild command line. The
+following properties change the behavior of `Grpc.Tools`:
+
+| Name                | Synopsis                                                                      |
+|---------------------|-------------------------------------------------------------------------------|
+| `Protobuf_ToolsOs`  | Same as `PROTOBUF_TOOLS_OS` environment variable                              |
+| `Protobuf_ToolsCpu` | Same as `PROTOBUF_TOOLS_CPU` environment variable                             |
+| `Protobuf_ProtocFullPath` | Same as `PROTOBUF_PROTOC` environment variable                          |
+| `gRPC_PluginFullPath` | Same as `GRPC_PROTOC_PLUGIN` environment variable                           |
+| `Protobuf_NoWarnMissingExpected` | Default: `false`. If `true` then no warnings are given if expected files not generated. See example below for an explanation. |
+| `Protobuf_OutputPath`| Default: `IntermediateOutputPath` - ususally the `obj` directory. Sets the default value for `OutputDir` on `<Protobuf>` items.|
+
+# Scenarios and Examples
+
+For other examples see also the `.csproj` files in the examples in GitHub:
+* Grpc-dotnet examples: https://github.com/grpc/grpc-dotnet/tree/master/examples
+* Grpc c-core C# examples: https://github.com/grpc/grpc/tree/v1.46.x/examples/csharp
+
+Quick links to the examples below:
+
+* [AdditionalImportDirs - Setting location of imported `.proto` files](#AdditionalImportDirs)
+* [ProtoRoot - Avoiding duplicate generated file names](#ProtoRoot)
+* [GrpcServices - Generating gRPC services and protocol buffers messages](#grpcservices)
+* [Generate proto and gRPC C# sources from .proto files (no C# compile)](#nocompile)
+* [Visual Studio: setting per-file `.proto` file options](#visualstudio)
+* [Bypassing Grpc.Tools to run the protocol buffers compiler explicitly](#compiler)
+
+---
+## <a name="AdditionalImportDirs"></a>AdditionalImportDirs - Setting location of imported `.proto` files
+
+If your `.proto` files import other files that are not co-located with the `.proto` file 
+you can specify the directories to search for the
+imported files by specifying `AdditionalImportDirs` and provide a list of directories.
+The directories are searched in the order given. E.g.:
+
+```xml
+  <Protobuf Include="protos/*.proto" ProtoRoot="protos"
+      AdditionalImportDirs="/folder/protos/myapis/;/another/folder/"
+      ... />
+```
+
+Note - the paths for standard imports included in the `Grpc.Tools` package are automatically passed to the protocol buffers compiler and do not need to be explicitly set.
+E.g. the directory `"...\.nuget\packages\grpc.tools\2.51.0\build\native\include"` is automatically included.  This is the directory where a declaration such as `import "google/protobuf/wrappers.proto";` in a proto file would be found.
+
+---
+## <a name="ProtoRoot"></a>ProtoRoot - Avoiding duplicate generated file names
```

Comment:
AFAIK the ProtoRoot is especially important for making sure that the  "import foo/bar/b.proto" statements will work as expected.  Basically, if you have `Protos/foo/bar/a.proto` and `Protos/foo/bar/b.proto` and a.proto imports b.proto via `import foo/bar/b.proto`, the ProtoRoot needs to be set to "Protos" otherwise the protoc won't see b.proto and build will fail.
This isn't something that's mentioned in the doc currently, but it's very important and people often get it wrong.